src/core/server/plugins/discovery/plugin_discovery.test.ts

```diff
@@ -0,0 +1,483 @@
+    },
+    logger.get()
+  );
+
+  await expect(plugin$.pipe(toArray()).toPromise()).resolves.toMatchInlineSnapshot(`
+Array [
+  Object {
+    "manifest": Object {
+      "id": "plugin",
+      "kibanaVersion": "1.2.3",
+      "optionalPlugins": Array [],
+      "requiredPlugins": Array [],
+      "ui": false,
+      "version": "1",
+    },
+    "path": "${resolveForSnapshot(TEST_PATHS.scanDirs.nonEmpty, '1')}",
+  },
+  Object {
+    "manifest": Object {
+      "id": "plugin",
+      "kibanaVersion": "1.2.3",
+      "optionalPlugins": Array [],
+      "requiredPlugins": Array [],
+      "ui": false,
+      "version": "1",
+    },
+    "path": "${resolveForSnapshot(TEST_PATHS.scanDirs.nonEmpty, '3')}",
+  },
+  Object {
+    "manifest": Object {
+      "id": "plugin",
+      "kibanaVersion": "1.2.3",
+      "optionalPlugins": Array [],
+      "requiredPlugins": Array [],
+      "ui": false,
+      "version": "1",
+    },
+    "path": "${resolveForSnapshot(TEST_PATHS.scanDirs.nonEmpty2, '6')}",
+  },
+  Object {
+    "manifest": Object {
+      "id": "plugin",
+      "kibanaVersion": "1.2.3",
+      "optionalPlugins": Array [],
+      "requiredPlugins": Array [],
+      "ui": false,
+      "version": "1",
+    },
+    "path": "${resolveForSnapshot(TEST_PATHS.paths.existentDir)}",
+  },
+  Object {
+    "manifest": Object {
+      "id": "plugin",
+      "kibanaVersion": "1.2.3",
+      "optionalPlugins": Array [],
+      "requiredPlugins": Array [],
+      "ui": false,
+      "version": "1",
+    },
+    "path": "${resolveForSnapshot(TEST_PATHS.paths.existentDir2)}",
+  },
+]
+`);
+
+  await expect(
+    error$
+      .pipe(
+        map(error => error.toString()),
+        toArray()
+      )
+      .toPromise()
+  ).resolves.toMatchInlineSnapshot(`
+Array [
+  "Error: ENOENT (missing-manifest, ${resolveForSnapshot(
+    TEST_PATHS.scanDirs.nonEmpty,
+    '2-no-manifest',
+    'kibana.json'
+  )})",
+  "Error: Plugin manifest must contain an \\"id\\" property. (invalid-manifest, ${resolveForSnapshot(
+    TEST_PATHS.scanDirs.nonEmpty,
+    '4-incomplete-manifest',
+    'kibana.json'
+  )})",
+  "Error: Unexpected token o in JSON at position 1 (invalid-manifest, ${resolveForSnapshot(
+    TEST_PATHS.scanDirs.nonEmpty2,
+    '5-invalid-manifest',
+    'kibana.json'
+  )})",
+  "Error: Plugin \\"plugin\\" is only compatible with Kibana version \\"1\\", but used Kibana version is \\"1.2.3\\". (incompatible-version, ${resolveForSnapshot(
+    TEST_PATHS.scanDirs.nonEmpty2,
+    '8-incompatible-manifest',
+    'kibana.json'
+  )})",
+  "Error: ENOENT (invalid-scan-dir, ${resolveForSnapshot(TEST_PATHS.scanDirs.nonExistent)})",
+  "Error: ${resolveForSnapshot(
+    TEST_PATHS.paths.nonDir
+  )} is not a directory. (invalid-plugin-dir, ${resolveForSnapshot(TEST_PATHS.paths.nonDir)})",
+  "Error: ENOENT (invalid-plugin-dir, ${resolveForSnapshot(TEST_PATHS.paths.nonExistent)})",
+]
+`);
+});
+
+describe('parsing plugin manifest', () => {
```

Comment:
Mind moving these to a test file for the plugin manifest parser?

Reply:
Sure, created tests before extracted parser into a separate file, but it makes sense to extract them as well now.